### ISSUE TITLE
Make human friendly document maps when custom ids are not given

### ISSUE THREAD

**Issue by samuel-belcastro** [ucfopen/Obojobo#628]
### tl;dr - Current Obojobo node identification schemes within a document make document analytics difficult for authors. Authors must create their own identification schemes or reason about autogenerated UUIDs to analyze document data.

Currently, Obojobo document authors have the ability to add unique ids to the nodes they place in their Obojobo documents. For example, an author adds an assessment node as follows:
```
<Assessment id="my-custom-id">
    <Question id="my-question-1">
    <Question id="my-question-2">
</Assessment>
```
The document engine forces custom ids to be unique on save. If an author has duplicate ids, the document will fail to save.

If the author does not provide an id, a UUID is generated and placed in the id property like so: 
```
<Assessment id="de98aa70-13ef-4796-94a6-e7fbb43babf3">
    <Question id="7a33b6e3-98fa-4bd5-9d21-14811234bd96">
    <Question id="2b9bd417-e086-4c45-ba26-d790937f87ea">
</Assessment>
```

The autogenerated UUIDs make querying document data more difficult. To query a tree of UUIDs, the user needs to manually map the node UUIDs to the nodes the user is trying to analyze.
````
| - de98aa70-13ef-4796-94a6-e7fbb43babf3
|   | - 7a33b6e3-98fa-4bd5-9d21-14811234bd96
|   | - 2b9bd417-e086-4c45-ba26-d790937f87ea
`````
To get *question one*  analytics the user needs to look at the document, find the UUID for question 1, and query that UUID. This is a tedious process, and becomes more difficult as the document grows. This process can be streamlined with the custom ids, but custom ids can be tedious as well. Each author must come up with a custom id scheme to apply to his or her documents.

### tl;dr - Create a visual map of the document and allow authors to click/select on the nodes they want to reason about instead of searching through the document for unique identifiers.

![screen shot 2018-11-07 at 10 45 02 am](https://user-images.githubusercontent.com/17661897/48142155-3e112a80-e27a-11e8-96c8-f76ca730cf21.png)

The above map shows a document based on the node names, but this will not always me the case. The map will describe each node based on the following rules:

| Custom id is provided? | Node title is provided? | Unique ID is generated? | Display |
|------------------------|------------------------|-------------------------|--------|
| Yes | Yes | Yes | Custom id |
| Yes | No | Yes | Custom id |
| No | Yes | Yes | Title |
| No | No | Yes | Node name |

Order of display precedence: 

`Custom id > Node title > Node name`
